metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+```
+:filter(and(eq(VisitorType,NEW_VISITOR),ne(VisitType,ROBOT)))
+```
+We observe that the payload has changed to only contain the result data we are interested in. For each application, two data points remain per timeslot, one for `SYNTHETIC`, and one for `REAL_USER`, while the visitor type is constant at `NEW_VISITOR`. We actually want the average of synthetic and real, so that we have one data point per application. We can easily view them as a combined data point by using a `merge` transformer. It takes the names or indexes of dimensions to remove as arguments, merging data points that have become equal when the dimension is removed:
+```
+:merge(VisitType,VisitorType)
+```
+Alternatively, we can select only the dimensions to show up in the result and merge the other dimensions. `:splitBy` can be used for an equivalent merge operation:
+```
+:splitBy(dt.entity.application)
+```
+Combining or `:filter` and our `:merge` transformation, we see that we have solved our problem and get the data in the exact format that we need it in:
+```
+metricId,dt.entity.application,time,value
+"builtin:apps.web.sessionDuration:filter(and(eq(VisitorType,NEW_VISITOR),ne(VisitType,ROBOT))):merge(VisitType,VisitorType)",APPLICATION-1AF167A3A7B45A8A,2019-04-09 00:00:00,20.51584226122635
+...
+```
+Note how the semantics of the transformer chain change with the ordering of the transformers. The filter does not make sense after the merge.
+
+We get an error since we just merged the visitor type and then tried to filter on the dimension we just removed.
+
+# Scenario 10: All Service Methods of a Service
+
+>**Task**
+>We want to get information for our web service, but by default the data is reported on service method level.
+
+Transformers allow us to shape a metric to our needs. Until now, we have done that by removing dimensions, or by removing result rows. Some transformers also add dimensions, and `:parents` is one of them.
+
+Consider the following problem: The metric `builtin:service.keyRequest.errors.fourxx` is recorded per service method. We are interested in the service methods, but we want to filter the result payload by the ID of the service, namely `SERVICE-0123456789ABCDEF`. We cannot use a scope, since the primary entity is `SERVICE_METHOD` and not `SERVICE`. A `:filter` transformation would work, if the metric contained a secondary service dimension, which it does not. It can easily be added, though, by prepending the transformer `:parents`.
+
+`:parents` is not only suitable for adding application entities. The general contract of `:parents` is that for each monitored entity dimension, a new dimension will be added before the existing one, if the entity is naturally embedded inside a larger entity of which there can only be one. Disks are a good example. The disk needs the host, otherwise monitoring is not possible. The following have similar disk-like relations that `:parents` is suitable for:

```

Comment:
```suggestion
`:parents` is not only suitable for adding application entities. The general contract of `:parents` is that for each monitored entity dimension, a new dimension will be added before the existing one, if the entity is naturally embedded inside a larger entity of which there can only be one. Disks are a good example. A disk needs a host, otherwise monitoring is not possible. The following have similar disk-host-like relations that `:parents` is suitable for:

```